### ISSUE TITLE
Allow hiding of toolchains

### ISSUE THREAD

**Issue by geimer** [hpcugent/easybuild-framework#1682]
While this request may seem odd at first glance, hiding the `GCCcore` toolchain may nevertheless be useful.  Having set `EASYBUILD_HIDE_DEPS=GCCcore` in my environment, executing `eb -D GCC-4.9.3-2.25.eb` shows
```
Dry run: printing build status of easyconfigs and dependencies
CFGS=.../easybuild-easyconfigs/easybuild/easyconfigs
 * [ ] $CFGS/m/M4/M4-1.4.17.eb (module: M4/1.4.17)
 * [ ] $CFGS/f/flex/flex-2.5.39.eb (module: flex/2.5.39)
 * [ ] $CFGS/b/Bison/Bison-3.0.4.eb (module: Bison/3.0.4)
 * [ ] $CFGS/z/zlib/zlib-1.2.8.eb (module: zlib/1.2.8)
 * [ ] $CFGS/b/binutils/binutils-2.25.eb (module: binutils/2.25)
 * [ ] $CFGS/g/GCCcore/GCCcore-4.9.3.eb (module: GCCcore/.4.9.3)
 * [ ] $CFGS/g/GCCcore/GCCcore-4.9.3.eb (module: GCCcore/4.9.3)
 * [ ] $CFGS/m/M4/M4-1.4.17-GCCcore-4.9.3.eb (module: M4/1.4.17-GCCcore-4.9.3)
 * [ ] $CFGS/f/flex/flex-2.5.39-GCCcore-4.9.3.eb (module: flex/2.5.39-GCCcore-4.9.3)
 * [ ] $CFGS/b/Bison/Bison-3.0.4-GCCcore-4.9.3.eb (module: Bison/3.0.4-GCCcore-4.9.3)
 * [ ] $CFGS/z/zlib/zlib-1.2.8-GCCcore-4.9.3.eb (module: zlib/1.2.8-GCCcore-4.9.3)
 * [ ] $CFGS/b/binutils/binutils-2.25-GCCcore-4.9.3.eb (module: binutils/2.25-GCCcore-4.9.3)
 * [ ] $CFGS/g/GCC/GCC-4.9.3-2.25.eb (module: GCC/4.9.3-2.25)
```
As one can see, `GCCcore` shows up twice in the list of packages to be built -- once hidden and once non-hidden.  The hidden build comes from the `GCC-4.9.3-2.25.eb` bundle which uses the `dummy` toolchain and lists `GCCcore` only as a dependency (where hiding works OK).  The non-hidden build is triggered by, e.g., `flex-2.5.39-GCCcore-4.9.3.eb` which uses `GCCcore` as a toolchain but not as a dependency.

I haven't checked yet what happens during building (i.e., whether `GCCcore` is really built twice), thus this may also be a bug in `--dry-run`.

**Comment by ocaisa** [comment]
I'm pretty sure that this would result with the build being done twice since EB checks against an available module and since it's not looking for an invisible module

**Comment by ocaisa** [comment]
BTW I'm going to town on `EASYBUILD_HIDE_DEPS` at the moment! Check this
```
-bash-4.2$ ma

------------------------------------------------------ /usr/local/software/jureca/Stages/2016a/modules/all/MPI/intel/2016.2.181-GCC-4.9.3-2.25/impi/5.1.3.181 -------------------------------------------------------
   Boost/1.60.0-Python-2.7.11    HDF5/1.8.16    SIONlib/1.6.1    imkl/11.3.2.181    ncview/2.1.6    netCDF/4.4.0    netCDF-C++4/4.2.1    netCDF-Fortran/4.4.3

--------------------------------------------------------- /usr/local/software/jureca/Stages/2016a/modules/all/Compiler/mpi/intel/2016.2.181-GCC-4.9.3-2.25 ----------------------------------------------------------
   IntelMPI/5.1.3.181    MVAPICH2/2.2b-GDR (g)    ParaStationMPI/5.1.5-1

----------------------------------------------------------- /usr/local/software/jureca/Stages/2016a/modules/all/Compiler/intel/2016.2.181-GCC-4.9.3-2.25 ------------------------------------------------------------
   CUDA/7.5.18 (g)    GMP/6.1.0    PyGTK/2.24.0-Python-2.7.11    PyOpenGL/3.1.1a1-Python-2.7.11    Python/2.7.11

-------------------------------------------------------------------- /usr/local/software/jureca/Stages/2016a/modules/all/Compiler/GCCcore/4.9.3 ---------------------------------------------------------------------
   Autotools/20150215    CMake/3.4.3    Doxygen/1.8.11    Grace/5.1.25    LLVM/3.7.1    PAPI/5.4.3    Perl/5.22.1    cURL/7.47.1    git/2.7.2    util-linux/2.27.1

----------------------------------------------------------------------------- /usr/local/software/jureca/Stages/2016a/modules/all/Core ------------------------------------------------------------------------------
   EasyBuild/2.6.0    GCC/5.3.0-2.26 (D)    GitPython/1.0.2                    PGI/16.1-GCC-4.9.3-2.25    gpsmpi/2016a                  imvapich2c/2016a-GDR    intel-para/2016a        pmvmklc/2016a-GDR
   GC3Pie/2.4.2       GCCcore/4.9.3         Intel/2016.2.181-GCC-4.9.3-2.25    gmvapich2c/2016a-GDR       gpsolf/2016a                  imvmklc/2016a-GDR       ipsmpi/2016a            rpmrebuild/2.11
   GCC/4.9.3-2.25     GCCcore/5.3.0  (D)    Java/1.8.0_74                      gmvolfc/2016a-GDR          iimpi/8.2.5-GCC-4.9.3-2.25    intel/2016a             pmvapich2c/2016a-GDR
```
compared to this
```
-bash-4.2$ module --show-hidden av

------------------------------------------------------ /usr/local/software/jureca/Stages/2016a/modules/all/MPI/intel/2016.2.181-GCC-4.9.3-2.25/impi/5.1.3.181 -------------------------------------------------------
   Boost/1.60.0-Python-2.7.11    HDF5/1.8.16    OTF2/.2.0    SIONlib/1.6.1    imkl/11.3.2.181    ncview/2.1.6    netCDF/4.4.0    netCDF-C++4/4.2.1    netCDF-Fortran/4.4.3

--------------------------------------------------------- /usr/local/software/jureca/Stages/2016a/modules/all/Compiler/mpi/intel/2016.2.181-GCC-4.9.3-2.25 ----------------------------------------------------------
   IntelMPI/5.1.3.181    MVAPICH2/2.2b-GDR (g)    ParaStationMPI/5.1.5-1

----------------------------------------------------------- /usr/local/software/jureca/Stages/2016a/modules/all/Compiler/intel/2016.2.181-GCC-4.9.3-2.25 ------------------------------------------------------------
   CUDA/7.5.18 (g)    PyCairo/.1.10.0-Python-2.7.11      PyGTK/2.24.0-Python-2.7.11        Python/2.7.11    UDUNITS/.2.2.20    zlib/.1.2.8
   GMP/6.1.0          PyGObject/.2.28.6-Python-2.7.11    PyOpenGL/3.1.1a1-Python-2.7.11    Szip/.2.1        bzip2/.1.0.6

-------------------------------------------------------------------- /usr/local/software/jureca/Stages/2016a/modules/all/Compiler/GCCcore/4.9.3 ---------------------------------------------------------------------
   ATK/.2.18.0                      Gdk-Pixbuf/.2.32.3    Perl/5.22.1           fixesproto/.5.0        libICE/.1.0.9        libXpm/.3.5.11          libpthread-stubs/.0.3        util-linux/.2.27.1
   Autoconf/.2.69                   Grace/5.1.25          Qt/.4.8.7             flex/.2.5.39           libSM/.1.2.2         libXrender/.0.9.9       libreadline/.6.3             util-linux/2.27.1
   Automake/.1.15                   HarfBuzz/.1.1.3       SQLite/.3.9.2         flex/.2.6.0            libX11/.1.6.3        libXt/.1.1.5            libtool/.2.4.6               wxWidgets/.3.1.0
   Autotools/20150215               LLVM/3.7.1            Tcl/.8.6.4            fontconfig/.2.11.94    libXau/.1.0.8        libdrm/.2.4.67          libunwind/.1.1               xcb-proto/.1.11
   Bison/.3.0.4                     LibTIFF/.4.0.6        Tk/.8.6.4             fontsproto/.2.1.3      libXaw/.1.0.12       libdwarf/.20160116      libxcb/.1.11.1               xextproto/.7.3.0
   CMake/3.4.3                      LibUUID/.1.0.3        binutils/.2.25        freetype/.2.6.2        libXdamage/.1.1.4    libelf/.0.8.13          libxml2/.2.9.3               xorg-macros/.1.19.0
   Coreutils/.8.25                  M4/.1.4.17            bzip2/.1.0.6          gettext/.0.19.7        libXdmcp/.1.1.2      libffi/.3.2.1           makedepend/.1.0.5            xproto/.7.0.28
   Doxygen/.1.8.11                  Mesa/.11.1.2          cURL/7.47.1           git/2.7.2              libXext/.1.3.3       libfontenc/.1.1.3       motif/.2.3.4-libX11-1.6.3    xtrans/.1.3.5
   Doxygen/1.8.11                   NASM/.2.11.08         cairo/.1.14.6         glproto/.1.4.17        libXfixes/.5.0.1     libglade/.2.6.4         ncurses/.6.0                 zlib/.1.2.8
   GLib/.2.47.5                     PAPI/5.4.3            damageproto/.1.2.1    gperf/.3.0.4           libXfont/.1.5.1      libjpeg-turbo/.1.4.2    pixman/.0.34.0
   GObject-Introspection/.1.47.1    PCRE/.8.38            eudev/.3.1.5          inputproto/.2.3.1      libXft/.2.3.2        libpciaccess/.0.13.4    pkg-config/.0.29
   GTK+/.2.24.28                    Pango/.1.39.0         expat/.2.1.0          kbproto/.1.0.7         libXmu/.1.1.2        libpng/.1.6.21          renderproto/.0.11

----------------------------------------------------------------------------- /usr/local/software/jureca/Stages/2016a/modules/all/Core ------------------------------------------------------------------------------
   Bison/.3.0.4       GCC/5.3.0-2.26                  (D)    Java/1.8.0_74              flex/.2.5.39            gpsolf/2016a                        imvmklc/2016a-GDR       pmvmklc/2016a-GDR    zlib/.1.2.8
   EasyBuild/.dev     GCCcore/4.9.3                          M4/.1.4.17                 flex/.2.6.0             icc/.2016.2.181-GCC-4.9.3-2.25      intel/2016a             popt/.1.16
   EasyBuild/2.6.0    GCCcore/5.3.0                   (D)    PGI/16.1-GCC-4.9.3-2.25    gmvapich2c/2016a-GDR    ifort/.2016.2.181-GCC-4.9.3-2.25    intel-para/2016a        pscom/.Default
   GC3Pie/2.4.2       GitPython/1.0.2                        binutils/.2.25             gmvolfc/2016a-GDR       iimpi/8.2.5-GCC-4.9.3-2.25          ipsmpi/2016a            pscom/.5.0.48-1
   GCC/4.9.3-2.25     Intel/2016.2.181-GCC-4.9.3-2.25        binutils/.2.26             gpsmpi/2016a            imvapich2c/2016a-GDR                pmvapich2c/2016a-GDR    rpmrebuild/2.11

```

**Comment by boegel** [comment]
fixed via #1683
